RocketChat/Rocket.Chat
# [FIX] New messages in threads are only marked as read after a message is sent in the main room

## Proposed changes (including videos or screenshots)


 - Update rooms' `lm` fields when messages in threads are sent.



## Issue(s)


## Steps to test or reproduce

1. Go to **Administration > Message** and enable "Show Read Receipts" and "Detailed Read Receipts";
2. Login with two users and access the same room with both of them;
3. As "User 1", send a message in an existing thread;
4. Access this room as "User 2". **_Expected behavior:_** the message sent in step 3 should be marked as read by "User 2" (even though the user hasn't opened the thread yet);
5. Open the thread as "User 2" and send a new message in the thread. **_Expected behavior:_** the new message should be instantly mark as read by both users.

## Further comments

By updating rooms' `lm` field when messages in threads are sent, the condition in [this line](https://github.com/RocketChat/Rocket.Chat/blob/9d818a7af44ecac02982085634cc2e0b2bdc095e/apps/meteor/imports/message-read-receipt/server/lib/ReadReceipt.js#L46) may not be met, which allows new read receipts to be created (since the new thread messages will be found in `findUnreadMessagesByRoomAndDate`).
